src/connect/ConnectWizard.ts

```diff
@@ -202,8 +204,31 @@ export class ConnectWizard {
             activeItem: serviceChoices[0]
         });
         this._result.resourceName = pick.label;
+        // get the list of containers
+        const podName = await kubectlClient.getPodName(this._result.resourceName);
+        const containersList: string[] = await kubectlClient.getContainersList(podName);
+        if (containersList?.length > 1) {
+            // show containers quick pick list after filtering known sidecar containers
+            const containerChoices: vscode.QuickPickItem[] = containersList.map(containers => ({ label: containers }));
+            return (input: MultiStepInput) => this.inputContainersAsync(input, containerChoices, resourceType);
+        } else {
```

Comment:
💡 so this `else` will always hit if `containersList` is `null` aka not satisfy the `if` clause first part? 

Also, what happened to this clause:

```*.ts
       if (!containersList || !containersList.length) {
            // containerList or containerlist.length are falsy
            // ⇒ do not attempt to process containerlist
            return (input: MultiStepInput) => this.inputPortsAsync(input, resourceType);
        }
```

The actual comment referring to is here: 

https://github.com/Azure/vscode-bridge-to-kubernetes/pull/71/files#r1329327043

Reply:
it is handled, if containersList length is zero or containersList is null it would go to the else condition, this._result.containerName would be undefined. I have handled if containerName is undefined or null or empty it wouldn't populate the value in the other parts of the code ex: see DebugAssestsInitializer.ts. I tested different scenarios and all of them worked, as well Unit tests covers those scenarios. if it is not readable I can change it back to simple one. I feel I might have over complicated this condition. :)

Reply:
This is updated as well.